h/views/api/config.py

```diff
@@ -51,24 +63,45 @@ def add_api_view(
         # preferred
         primary_method = primary_method[0]
 
-    settings.setdefault("accept", "application/json")
     settings.setdefault("renderer", "json")
     settings.setdefault("decorator", cors_policy)
 
-    if link_name:
-        link = {
-            "name": link_name,
-            "method": primary_method,
-            "route_name": settings.get("route_name"),
-            "description": description,
-        }
-
-        registry = config.registry
-        if not hasattr(registry, "api_links"):
-            registry.api_links = []
-        registry.api_links.append(link)
+    current_version = config.registry.settings["api.version.current"]
+    known_versions = config.registry.settings["api.versions"]
+    view_versions = versions or [current_version]
+
+    registry = config.registry
+    if not hasattr(registry, "api_links"):
+        registry.api_links = {}
+
+    for version in view_versions:
+        if version not in known_versions:
+            raise APIConfigError(
```

Comment:
I didn't bother formatting/defining this Exception carefully yet until I'm sure what kind of Exception-raising we wanna do here.